### DISCUSSION TITLE
Is there any cluster node Health Check API?

### DISCUSSION THREAD

**Discussion by heartcored98** [rancher/rke2#2837] [General]
**Is your feature request related to a problem? Please describe.**

It is not related to a problem but it is rather related to feature request of question. 
I am trying to operate on-premise RKE2 cluster consist of 3 nodes. 
Those nodes are behind the subdomain which hosted on Route53 with weighted routing policy in order to distribute the traffic across the three nodes. (I recognized that using Route53 as load balancer is not the optimal solution, but I think it is okay with me now.) 

I am looking for the way to implement Route53 DNS failover function with AWS Cloudwatch's custom metrics in order to exclude unhealthy node from the Route53 record. 
In order to achieve above goal, I am looking forward the RKE2 Node Health Check API. 

By checking node health check api periodically and publishing custom metrics to CloudWatch, I can automatically deactivate the Route53 A record when the corresponding RKE2 node went wrong. 

Therefore, I have searched for HTTP api for RKE2 node health check API, but I cannot find. 
Is there any way to simply check the node status? 

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by brandond** [comment]
What are you trying to health check? The control-plane, the ingress/load-balancer, or access to your workload through the ingress or load-balancer?

**Comment by heartcored98** [comment]
@brandond Hi, I am hoping to check the status of the control-plane.

**Reply by brandond** [reply]
In that case I would probably look at the upstream docs for the components you're hoping to health check - the apiserver, controller-manager, etc.

https://kubernetes.io/docs/reference/using-api/health-checks/

**Reply by heartcored98** [reply]
@brandond That's what I was looking for! You saved my time :)
